src/lib.rs

```diff
@@ -1463,6 +1463,27 @@ impl<K, V, S> Default for OrderMap<K, V, S>
     }
 }
 
+impl<K, V, S> PartialEq for OrderMap<K, V, S>
+    where K: Eq + Hash,
```

Comment:
I'd like to do this more generally than libstd.

We sometimes have room to fix their mistakes or oversights, right? And we should do it if it remains a drop in replacement.

What I first thought of here is that PartialEq should not require that both maps have the same hasher. But in fact, we can generalize it so that neither K, V, S need to be the same as long as they implement `V1: PartialEq<V2>` and so on. Does that make sense to you?

For `K` it's a bit trickier, we need to be able to look up using one key in the other map. I'm not sure what works there without trying it out in the code.